google/cloud/bigquery/_helpers.py

```diff
@@ -1007,59 +1007,50 @@ def _verify_job_config_type(job_config, expected_type, param_name="job_config"):
         )
 
 
-class ResourceBase:
-    """Base class providing the from_api_repr method."""
-
-    def __init__(self):
-        self._properties = {}
-
-    @classmethod
-    def from_api_repr(cls, resource: dict):
-        """Factory: constructs an instance of the class (cls)
-        given its API representation.
-
-        Args:
-            resource (Dict[str, Any]):
-                API representation of the object to be instantiated.
-
-        Returns:
-            ResourceBase: An instance of the class initialized with data
-                from 'resource'.
-        """
-        config = cls()
-        config._properties = copy.deepcopy(resource)
-        return config
-
-
-def _isinstance_or_raise(value: Any, dtype: Optional[Union[Any, Tuple[Any, ...]]]):
+def _isinstance_or_raise(
+    value: Any,
+    dtype: Union[Type, Tuple[Type, ...], None],
```

Comment:
Where do we use this with dtype=None? If nowhere, can we remove it?

```suggestion
    dtype: Union[Type, Tuple[Type, ...]],
```